chemprop/data/utils.py

```diff
@@ -333,98 +373,166 @@ def get_data(path: str,
         gt_targets, lt_targets = None, None
 
     # Load data
-    with open(path) as f:
+    extension = os.path.splitext(path)[1]
+    if extension == '.csv':
+        f = open(path)
         reader = csv.DictReader(f)
         fieldnames = reader.fieldnames
-        if any([c not in fieldnames for c in smiles_columns]):
-            raise ValueError(f'Data file did not contain all provided smiles columns: {smiles_columns}. Data file field names are: {fieldnames}')
-        if any([c not in fieldnames for c in target_columns]):
-            raise ValueError(f'Data file did not contain all provided target columns: {target_columns}. Data file field names are: {fieldnames}')
+    elif extension in ['.pkl', '.pckl', '.pickle']:
+        reader = pd.read_pickle(path)
+        fieldnames = reader.columns.tolist()
+        reader = reader.iterrows()
+
+    if any([c not in fieldnames for c in smiles_columns]):
+        raise ValueError(f'Data file did not contain all provided smiles columns: {smiles_columns}. Data file field names are: {fieldnames}')
+    if any([c not in fieldnames for c in target_columns]):
+        raise ValueError(f'Data file did not contain all provided target columns: {target_columns}. Data file field names are: {fieldnames}')
+
+    all_smiles, all_targets, all_atom_targets, all_bond_targets, all_rows, all_features, all_phase_features, all_constraints_data, all_raw_constraints_data, all_weights, all_gt, all_lt = [], [], [], [], [], [], [], [], [], [], [], []
+    for i, row in enumerate(tqdm(reader)):
+        targets, atom_targets, bond_targets = [], [], []
 
-        all_smiles, all_targets, all_rows, all_features, all_phase_features, all_weights, all_gt, all_lt = [], [], [], [], [], [], [], []
-        for i, row in enumerate(tqdm(reader)):
+        if extension == '.csv':
             smiles = [row[c] for c in smiles_columns]
 
-            targets = []
             for column in target_columns:
                 value = row[column]
-                if value in ['','nan']:
+                if value in ['', 'nan']:
                     targets.append(None)
                 elif '>' in value or '<' in value:
                     if loss_function == 'bounded_mse':
                         targets.append(float(value.strip('<>')))
                     else:
                         raise ValueError('Inequality found in target data. To use inequality targets (> or <), the regression loss function bounded_mse must be used.')
+                elif '[' in value or ']' in value:
+                    target = np.array(eval(value))
```

Comment:
It is generally a bad idea to use `eval` since it allows execution of arbitrary code. I'm not sure of the exact data structure expected here, but it may be possible to use `np.fromstring`.

Reply:
Thanks! This function is what I am looking for :)

Reply:
maybe `np.array(json.loads(value))`? `np.fromstring` doesn't expect `python` `list` syntax

Reply:
I tried to solve this problem by using regular expression but `np.array(json.loads(value))` is more elegant. Thanks for this suggestion!